src/panels/config/integrations/integration-panels/zha/zha-config-dashboard.ts

```diff
@@ -100,6 +157,45 @@ class ZHAConfigDashboard extends LitElement {
     `;
   }
 
+  private async _fetchConfiguration(): Promise<void> {
+    this._configuration = await fetchZHAConfiguration(this.hass!);
+  }
+
+  private _dataChangedCallback(section: string) {
+    return (ev: CustomEvent) => {
+      this._configuration!.data[section] = ev.detail.value;
+    };
+  }
+
+  private async _updateConfiguration(): Promise<any> {
+    const data = {};
+    for (const section of Object.keys(this._configuration!.schemas)) {
+      data[section] = {};
+      const sectionData = this._configuration!.data[section];
+      const sectionSchema = this._configuration!.schemas[section];
+      for (const field of sectionSchema) {
+        if (
+          field.name in sectionData &&
```

Comment:
Why dont we iterate over `sectionData` instead then?

Reply:
its an object and would need more object.keys junk and the schema is already iterable :) I can change it if you want.

Reply:
Is it because you need that default? Why do we need that default?